java/fury-core/src/main/java/org/apache/fury/resolver/XtypeResolver.java

```diff
@@ -0,0 +1,485 @@
+        xtypeId = getClassInfo(enclosingClass).xtypeId;
+      } else {
+        throw new ClassUnregisteredException(cls);
+      }
+    }
+    ClassInfo info = newClassInfo(cls, serializer, (short) xtypeId);
+    classInfoMap.put(cls, info);
+    return info;
+  }
+
+  private void registerDefaultTypes() {
+    registerDefaultTypes(Types.BOOL, Boolean.class, boolean.class, AtomicBoolean.class);
+    registerDefaultTypes(Types.INT8, Byte.class, byte.class);
+    registerDefaultTypes(Types.INT16, Short.class, short.class);
+    registerDefaultTypes(Types.INT32, Integer.class, int.class, AtomicInteger.class);
+    registerDefaultTypes(Types.INT64, Long.class, long.class, AtomicLong.class);
+    registerDefaultTypes(Types.FLOAT32, Float.class, float.class);
+    registerDefaultTypes(Types.FLOAT64, Double.class, double.class);
+    registerDefaultTypes(Types.STRING, String.class, StringBuilder.class, StringBuffer.class);
+    registerDefaultTypes(Types.DURATION, Duration.class);
+    registerDefaultTypes(
+        Types.TIMESTAMP,
+        Instant.class,
+        Date.class,
+        java.sql.Date.class,
+        Timestamp.class,
+        LocalDateTime.class);
+    registerDefaultTypes(Types.DECIMAL, BigDecimal.class, BigInteger.class);
+    registerDefaultTypes(
+        Types.BINARY,
+        byte[].class,
+        Platform.HEAP_BYTE_BUFFER_CLASS,
+        Platform.DIRECT_BYTE_BUFFER_CLASS);
+    registerDefaultTypes(Types.BOOL_ARRAY, boolean[].class);
+    registerDefaultTypes(Types.INT16_ARRAY, short[].class);
+    registerDefaultTypes(Types.INT32_ARRAY, int[].class);
+    registerDefaultTypes(Types.INT64_ARRAY, long[].class);
+    registerDefaultTypes(Types.FLOAT32_ARRAY, float[].class);
+    registerDefaultTypes(Types.FLOAT64_ARRAY, double[].class);
+    registerDefaultTypes(Types.LIST, ArrayList.class, Object[].class);
+    registerDefaultTypes(Types.SET, HashSet.class, LinkedHashSet.class);
+    registerDefaultTypes(Types.MAP, HashMap.class, LinkedHashMap.class);
+    registerDefaultTypes(Types.LOCAL_DATE, LocalDate.class);
+  }
+
+  private void registerDefaultTypes(int xtypeId, Class<?> defaultType, Class<?>... otherTypes) {
+    ClassInfo classInfo =
+        newClassInfo(defaultType, classResolver.getSerializer(defaultType), (short) xtypeId);
+    classInfoMap.put(defaultType, classInfo);
+    xtypeIdToClassMap.put(xtypeId, classInfo);
+    for (Class<?> otherType : otherTypes) {
+      classInfo = newClassInfo(otherType, classResolver.getSerializer(otherType), (short) xtypeId);
+      classInfoMap.put(otherType, classInfo);
+    }
+  }
+
+  public ClassInfo writeClassInfo(MemoryBuffer buffer, Object obj) {
+    ClassInfo classInfo = getClassInfo(obj.getClass(), classInfoCache);
+    int xtypeId = classInfo.getXtypeId();
+    byte internalTypeId = (byte) xtypeId;
+    buffer.writeVarUint32Small7(xtypeId);
+    switch (internalTypeId) {
+      case Types.NAMED_ENUM:
+      case Types.NAMED_STRUCT:
+      case Types.NAMED_EXT:
```

Comment:
Lack of support for:
NAMED_POLYMORPHIC_STRUCT
NAMED_COMPATIBLE_STRUCT
NAMED_POLYMORPHIC_COMPATIBLE_STRUCT
NAMED_POLYMORPHIC_EXT

Reply:
good catch, fixed